kuma/javascript/src/header/main-menu.jsx

```diff
@@ -0,0 +1,254 @@
+//@flow
+import * as React from 'react';
+import { useMemo, useEffect } from 'react';
+
+import { getLocale, gettext } from '../l10n.js';
+
+export default function MainMenu(mdnDocument: Object) {
```

Comment:
@peterbe Switched back to `Object` here as using `DocumentData` causes a ton of Flow exceptions at [this line](https://github.com/mozilla/kuma/pull/5560/files#diff-6b097a971a343a594f429782869d3df7R99). The reason those exceptions happen is that Flow treats all of the properties of `DocumentData` as mandatory. Therefore we would have to mark all properties as optional in order to use it here. If all properties are not actually optional, we then lose valuable checking from Flow. Perhaps I am overlooking something else, but that is what it seems like to me.